go/pkg/client/tree.go

```diff
@@ -360,15 +379,27 @@ func buildTree(files map[string]*fileSysNode) (*treeNode, error) {
 	return root, nil
 }
 
-func packageTree(t *treeNode, stats *TreeStats) (root digest.Digest, blobs map[digest.Digest]*uploadinfo.Entry, err error) {
+// If tree is not nil, it will be populated with a flattened tree of path->digest.
+// prefix should always be provided as an empty string which will be used to accumolate path prefixes during recursion.
```

Comment:
Wouldnt a better pattern be to have a non recursive top level 
`func packageTree(t *treeNode, stats *TreeStats, tree map[string]digest.Digest) (root digest.Digest, blobs map[digest.Digest]*uploadinfo.Entry, err error)`
that calls a recursive
`func packageTreeWithPrefix(t *treeNode, stats *TreeStats, prefix string, tree map[string]digest.Digest) (root digest.Digest, blobs map[digest.Digest]*uploadinfo.Entry, err error)`

Reply:
It's usually useful to simplify the public interface. Considering this is internal and actually have a single call site, I thought it would be verbose for no real benefit.

Reply:
That's fair